### ISSUE TITLE
Question about GSEA. How to rank?

### ISSUE THREAD

**Issue by Sirin24** [satijalab/seurat#9274]
Hello, how to rank genes for GSEA? is the avg_log2fc sufficient to rank the genes for GSEA? On bioconductor support pay they advice against filter out by pvalue, meaning we should keep all genes. so how to rank them? I am familiar with performing the t.test on h5ad objects and using the scores but with seurat i am not sure how to proceed. Thank you.

**Comment by shijianlvxing-lab** [comment]
Yes, all genes should be preserved, but they should be sorted from largest to smallest based on avg_log2fc.

**Comment by Sirin24** [comment]
I am not sure as Seurat findmarkers() has had some changes so I would appreciate it if someone from satijalab could help me.

**Comment by samuel-marsh** [comment]
So you aren't going to want to just use avg_log2fc in the absence of significance measure because genes with log fold changes above a certain cutoff but not significant is not what you want to be testing.

Also presuming that you ran cell-level DE with FIndMarkers you also are not going to want to use those pvalues because they are not statistically valid due to double dipping.

In the meantime though I'm going to move this issue to Discussions as it is not about the Seurat package but about analysis using a different package.  Hopefully more from community can then weigh in for you.

Best,
Sam
